tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/PullRequestLabelFolderCapability.cs

```diff
@@ -0,0 +1,127 @@
+﻿using CreateRuleFabricBot.Rules.PullRequestLabel;
+using OutputColorizer;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace CreateRuleFabricBot.Rules.IssueRouting
+{
+    public class PullRequestLabelFolderCapability : BaseCapability
+    {
+        private static readonly string s_template = @"
+{
+      ""taskType"": ""trigger"",
+      ""capabilityId"": ""PrAutoLabel"",
+      ""subCapability"": ""Path"",
+      ""version"": ""1.0"",
+      ""id"": ""###taskId###"",
+      ""config"": {
+        ""configs"": [
+            ###labelConfig###
+        ],
+      ""taskName"" :""Auto PR based on folder paths ""
+      }
+    }
+";
+
+        private static readonly string s_configTemplate = @"
+          {
+            ""label"": ""###label###"",
+            ""pathFilter"": [ ###srcFolders### ],
+            ""exclude"": [ ]
+          }
+";
+
+        private readonly string _repo;
+        private readonly string _owner;
+        private readonly string _codeownersFile;
+
+        public PullRequestLabelFolderCapability(string org, string name, string codeownersFile)
```

Comment:
How often to we expect codeowners files to be updated? Does that only happen everytime the bot is restarted?

Reply:
This is a tool that we will run whenever we want to in order to update the rule. I would say that we should run it every time the code owners file is updated. We can start doing this manually and see if we want to automate it.

Reply:
Yeah it would be nice to have it update whenever a codeowners file is updated but I don't know a great place have that triggered. @mitchdenny any thoughts on where to potentially do that?